hamilflow/models/harmonic_oscillator/initial_conditions.py

```diff
@@ -0,0 +1,39 @@
+import math
+from typing import Any, Dict, cast
+
+from pydantic import BaseModel, Field
+
+
+class HarmonicOscillatorIC(BaseModel):
+    """The initial condition for a harmonic oscillator
+
+    :cvar x0: initial displacement
+    :cvar v0: initial velocity
+    :cvar phi: initial phase
+    """
+
+    x0: float = Field(default=1.0)
+    v0: float = Field(default=0.0)
+    phi: float = Field(default=0.0)
+
+
+def parse_ic_for_sho(omega: float, **kwargs: Any) -> Dict[str, float]:
```

Comment:
This one looks quite bad. Can we reduce the complexity by settle on one certain kind of convention?

Reply:
Our goal is not to generate a good interface for users to use. As long as we can set the convention for ourselves, I am okay with anyone of the three choices.

Reply:
Note that $(E, t_0)$ is associated with the first integral of the system. First integrals, or constants of motion, have special and physical meaning in the framework of analytical mechanics. This is in contrast with the phase $\phi$.

Let us sleep over it a bit and then decide.

Reply:
> Note that (E,t0) is associated with the first integral of the system. First integrals, or constants of motion, have special and physical meaning in the framework of analytical mechanics. This is in contrast with the phase ϕ.
> 
> Let us sleep over it a bit and then decide.

Can we just keep $(E, t_0)$ and drop the others?